include/drivers/ipm.h

```diff
@@ -87,12 +87,137 @@ typedef void (*ipm_register_callback_t)(const struct device *port,
  */
 typedef int (*ipm_set_enabled_t)(const struct device *ipmdev, int enable);
 
+/**
+ * @brief Provides a type to hold IPM channel IDs.
+ *
+ * This type is sufficient to hold all the possible channels of the IPM peripheral
+ */
+typedef uint8_t ipm_channel_id_t;
```

Comment:
Some hardware peripherals used to implement ipm driver can utilize 32-bit register to pass "channel id" between cores. It could be better to extend this type to `uint32_t`

Reply:
ACK